casper/src/main/resources/PoS.rhox

```diff
@@ -133,12 +135,14 @@ new  PoS,
                   getInvalidBlocks(`rho:casper:invalidBlocks`) in {
                 getInvalidBlocks!(*invalidBlocksCh) |
                 getUser!(*userCh) |
-                for (@invalidBlocks <- invalidBlocksCh; @(rewards, bonds) <- stateCh; @userPk <- userCh) {
+                for (@invalidBlocks <- invalidBlocksCh; @(pending, rewards, bonds) <- stateCh; @userPk <- userCh) {
                   new toBeSlashed in {
                     toBeSlashed!(invalidBlocks.getOrElse(blockHash, userPk)) |
                     for (@validator <- toBeSlashed) {
                       // TODO: Transfer to coop wallet instead of just simply setting bonds to 0
-                      stateCh!((rewards.set(validator, 0), bonds.set(validator, 0))) |
+                      stateCh!((pending.set(validator, 0),
```

Comment:
Why does slashing not result in the validator being removed entirely from the `pending`, `rewards`, and `bonds` maps? Why are we wasting storage keeping track of the zero balances?

Reply:
https://github.com/rchain/rchain/blob/dev/casper/src/main/scala/coop/rchain/casper/Validate.scala#L741

Reply:
@KentShikama I don't see where `getBonds` in the `PoS` contract is accessed in that function.

Reply:
https://github.com/rchain/rchain/blob/dev/casper/src/main/scala/coop/rchain/casper/Validate.scala#L761

Reply:
It is cached and validated once. And that cached valued in the block is used.

Reply:
@KentShikama looking at https://github.com/rchain/rchain/blob/dev/casper/src/main/scala/coop/rchain/casper/Validate.scala#L741 the result is the same if the stake is zero or missing from the bonds map. Does this mean that, at slashing, we can remove the entry from the bonds map?